dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -17,7 +18,10 @@ function DatePickerWidget({
   setDateRange,
 }) {
   const [fromDate, setFromDate] = useState({});
-  const [toDate, setToDate] = useState(new Date());
+  const [toDate, setToDate] = useState(
+    constructUTCDate(new Date(formatDate(new Date())))
+  );
+  const [strDate, setStrDate] = useState(formatDate(new Date()));
```

Comment:
I would avoid using a local variable for strDate,since local variable values does not persist across re-renders,so we might get in trouble(maybe now,maybe in future,maybe we won't).Using state seems like a more solid approach to me,though if there is any reason not to use it as state,I am all ears and would love to make the desired change

Reply:
@shubham-html-css-js , I need you to educate me on the effects of a re-render.  Won't that cause the `DatePickerWidget()` function to be called again, which will establish new values/bindings for all of the local "variables" (like `toDate`)?  If that's the case, the we can just initialize `strDate` with the value of `toDate`, and it should pick up the right value, no?

And, we won't need to update `strDate` in `onFromChange`, because the effects of that callback won't be visible until after the next re-render, at which point `strDate` will pick up the value from `toDate` on the next call.

No?